system/lib/wasmfs/file.h

```diff
@@ -65,10 +64,11 @@ class File : public std::enable_shared_from_this<File> {
 
 protected:
   File(FileKind kind) : kind(kind) {}
+  File(FileKind kind, uint32_t mode)
+    : kind(kind), mode(mode), ctime(time(0)), mtime(time(0)), atime(time(0)) {}
```

Comment:
The other fields don't need to be initialized here because they are already initialized via the `= 0`.
```suggestion
    : kind(kind), mode(mode) {}
```

Also, it would be good to update the comment on `mode` below to clarify that it refers to user and group access control permissions on the file rather than what operations the file supports. What does the current JS system do with these permissions? Can we just ignore them for now?

Reply:
I think the current file system uses them to check for access:

https://github.com/emscripten-core/emscripten/blob/6cb821b7e35c39155ed4bce7fdb940c7f622ca86/src/library_fs.js#L324-L336

Reply:
So it looks like access is granted if any of the user, group, or other permissions allow for it :+1: